library/core/src/any.rs

```diff
@@ -1034,6 +1034,89 @@ impl<'a> Demand<'a> {
     pub fn is_satisfied(&self) -> bool {
         self.0.is_satisfied()
     }
+
+    /// Check if the `Demand` would be satisfied if provided with the
+    /// specified type. If the type does not match or has already been
+    /// provided, returns false.
```

Comment:
For some reason I can't suggest an edit, but I think the first sentence should be edited to say "would be satisfied if provided with a value of the specified type" and the `would_be_satisfied_by_ref_of` method should say "if provided with a reference to a value the specified type" or something along those lines